src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -169,6 +177,35 @@ func useAllCores() {
 	runtime.GOMAXPROCS(numcores)
 }
 
+func configureLogging() *rotator.SizeRotator {
+	logdir := appdir.Logs("Lantern")
+	log.Debugf("Placing logs in %v", logdir)
+	if _, err := os.Stat(logdir); err != nil {
+		if os.IsNotExist(err) {
+			// Create log dir
+			if err := os.MkdirAll(logdir, 0755); err != nil {
+				log.Fatalf("Unable to create logdir at %s: %s", logdir, err)
+			}
+		}
+	}
+	file := rotator.NewSizeRotator(filepath.Join(logdir, "lantern.log"))
+	// Set log files to 1 MB
+	file.RotationSize = 1 * 1024 * 1024
+	// Keep up to 20 log files
+	file.MaxRotation = 20
+	errorOut := timestamped(io.MultiWriter(file, os.Stderr))
+	debugOut := timestamped(io.MultiWriter(file, os.Stdout))
+	golog.SetOutputs(errorOut, debugOut)
+	return file
+}
+
+// timestamped adds a timestamp to the beginning of log lines
+func timestamped(orig io.Writer) io.Writer {
+	return wfilter.Lines(orig, func(w io.Writer, line string) (int, error) {
+		return fmt.Fprintf(w, "%s - %v", time.Now().Format(time.RFC3339), line)
```

Comment:
I chose the formatter that I did because it is constant width and preserves alignment of the log messages, which helps readability.

I do like the more compact representation, so I am using a fixed-width version of it:

`Jan 02 15:04:05.000`

I'm also converting the time to UTC before logging it so that we don't have to bother including a timezone.  This is important because we'll be getting log messages from all over the world.